File: 4/richard_zhan/simpleimg.py
import math
s = "P3 500 500 255\n"
for y in range(500):
    for x in range(500):
        center = (((x+10)/50)*50,((y+10)/50)*50)
        if (math.pow(center[0]*1.0-x,2) + math.pow(center[1]*1.0-y,2) <= 50):
                s += "255 255 255 "
        elif y % 50 >= 45 or y % 50 <= 5 or x % 50 >= 45 or x % 50 <= 5:
            s += "176 176 176 "
        else:
            s += "0 0 0 "
        
    
f = open("coolimg.ppm","w")
f.write(s)

File: 4/richard_zhan/test/simpleimg.c
#include<stdio.h>
#include<stdlib.h>
#include<fcntl.h>
#include<unistd.h>
#include<string.h>
#include<math.h>
int main(){
  int fd = open("coolimg.ppm",O_WRONLY | O_APPEND | O_CREAT , 0600);
  write(fd,"P3 500 500 255\n",strlen("P3 500 500 255\n"));
  int y = 0;
  int x = 0;
  for (y = 0; y < 500; y++){
    for (x = 0; x < 500; x++){
      //will make fancy later
      //char stuff[12] = "255 0 0 ";
      char stuff[12];
      sprintf(stuff,"%d %d %d ",y%255, x%255, 255);
      write(fd,stuff,strlen(stuff));
    }
  }
  close(fd);
  return 0;
}
